internal/controller/manager/controllermanager.go

```diff
@@ -136,41 +139,12 @@ func (c *ControllerManager) AddGroup(namespace string) error {
 // namespace specified, and then immediately runs the controllers in that group.
 func (c *ControllerManager) AddAndRunGroup(namespace string) error {
 
-	if c.controllers[namespace] != nil {
-		// first try to clean if one is not already in progress
-		if err := c.clean(namespace); err != nil {
-			log.Infof("Controller Manager: %s", err.Error())
-		}
-
-		// if we just cleaned the current namespace's controller, then return
-		if _, ok := c.controllers[namespace]; !ok {
-			log.Infof("Controller Manager: controller group for namespace %s has already "+
-				"been cleaned", namespace)
-			return nil
-		}
-
-		// check if we can create RBAC in the namespace in order to the reconcile RBAC
-		// as needed to ensure proper operator functionality
-		canCreateRBACInNamespace, err := ns.CanCreateRBACInNamespace(
-			c.controllers[namespace].kubeClientset,
-			namespace, c.namespaceOperatingMode)
-		if err != nil {
-			return err
-		}
-
-		log.Debugf("Controller Manager: canCreateRBACInNamespace is '%t' for namespace %s",
-			canCreateRBACInNamespace, namespace)
-
-		// now reconcile RBAC in the namespace if allowed
-		if canCreateRBACInNamespace {
-			if err := ns.ReconcileTargetRBAC(c.controllers[namespace].kubeClientset, c.pgoNamespace,
-				namespace); err != nil {
-				return err
-			}
-		}
+	if c.controllers[namespace] != nil && !c.pgoConfig.Pgo.DisableReconcileRBAC {
+		// first reconcile RBAC in the target namespace if RBAC reconciliation is enabled
+		c.reconcileRBAC(namespace)
```

Comment:
This will be called on each namespace refresh interval to ensure all RBAC is properly reconciled.

Reply:
When `c.controllers[namespace] != nil`, `c.reconcileRBAC(namespace)` is called twice. Why is that?

Reply:
> When `c.controllers[namespace] != nil`, `c.reconcileRBAC(namespace)` is called twice. Why is that?

No, I see... this runs even if `addControllerGroup()` returns `ErrControllerGroupExists`...

Something about this being in two places still seems off. Shouldn't there be one place in the control loop where we make this choice?

Reply:
It's really just an effort to reconcile RBAC as quickly as possible.  So while we don't necessarily need to do this when adding the controller group since on the next refresh interval it will reconcile the RBAC anyway (which is essentially what always happens in `readonly` or `disabled` modes since the create namespace event comes in before the RBAC resources exist within it), it just provides a means to reconcile these resources right away when we can (e.g. when using `dynamic` and we have the ClusterRole needed to do so).

Also worth noting that on each refresh interval I do want the RBAC reconcile attempt to happen before we check to ensure the controller is running, so that each worker can reconcile the current namespace concurrently with any other workers prior to running into the mutex below.